simulator/mips/mips_register/t/unit_test.cpp

```diff
@@ -11,39 +17,101 @@
 #define GTEST_ASSERT_NO_DEATH(statement) \
     ASSERT_EXIT({{ statement } ::exit(EXIT_SUCCESS); }, ::testing::ExitedWithCode(0), "")
 
+static_assert(MIPSRegister::MAX_REG >= 32);
 
-
-TEST( MIPS_registers, Process_Correct_Args_Of_Constr)
+TEST( MIPS_registers, Args_Of_Constr)
 {
     // Call a constructor
-    GTEST_ASSERT_NO_DEATH( MIPSRegister reg( 34); );
+    for ( size_t i = 0; i < 32; ++i)
+    {
+        GTEST_ASSERT_NO_DEATH( MIPSRegister reg( i); );
+    }
+
+    // Wrong parameter
+    ASSERT_EXIT( MIPSRegister reg( 32), ::testing::ExitedWithCode( EXIT_FAILURE), "ERROR: Invalid MIPS register id*");
```

Comment:
The idea is to create MIPSRegister's from only two sources:
1. Instruction decoding (5 bits)
2. Predefined values (HI, LO, RA, Zero etc.)

So I test that passing '32' (6 bits) crashes simulator. To get 100% sure, we may pass all values we can keep in 8 bits, which is quite excessive. The best approach is to test values which may occur in our simulation environment. like 0xFF (-1), 0x7F (MAX_INT8), NO_VAL8 etc.

Reply:
I think the better and more useful exercise is to copy these unit tests and modify them for RISCVRegister class to handle its semantic (no hi/lo, return address is 0x1 instead of 0x31 etc.). After tests are established, you may implement RISCVRegister class itself.